src/graphnet/models/model.py

```diff
@@ -132,3 +132,8 @@ def load_state_dict(self, path: str) -> 'Model':  # pylint: disable=arguments-di
 
     def _get_batch_size(self, data: Data) -> int:
         return torch.numel(torch.unique(data.batch))
+
+    def inference(self):
```

Comment:
I figure we should also have functionality to switch back from inference mode to train/eval mode. This could be done by overriding the pytorch built-in `.train()` and `.eval()` methods (see https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.train and https://pytorch.org/docs/stable/generated/torch.nn.Module.html#torch.nn.Module.eval) to switch the internal inference "flag" to "off", and then calling the corresponding base class method (i.e. `super().train()` and `super().eval()`).

Reply:
Unfortunately, I was unable to implement this completely, since `.eval()` is called by pytorch when predicting from somewhere deep inside the framework that we can't easily override. (I have a dump of the exact location and the function calls that lead to it.)

But I think the idea is good enough to still be valid only for `.train()`, so I have implemented it there.

Reply:
I would be interested to better understand the differences between `.train()` and `.eval()` but for the purposes of this PR I think the proposed change is perfectly fine.